libs/zzz/page-optimize-pando/src/CharCalcProvider.tsx

```diff
@@ -0,0 +1,126 @@
+import { notEmpty, toDecimal } from '@genshin-optimizer/common/util'
+import { CalcContext } from '@genshin-optimizer/game-opt/formula-ui'
+import { constant } from '@genshin-optimizer/pando/engine'
+import type {
+  DiscMainStatKey,
+  DiscSetKey,
+  DiscSubStatKey,
+} from '@genshin-optimizer/zzz/consts'
+import {
+  getDiscMainStatVal,
+  getDiscSubStatBaseVal,
+} from '@genshin-optimizer/zzz/consts'
+import type {
+  CharOpt,
+  ICachedCharacter,
+  ICachedDisc,
+} from '@genshin-optimizer/zzz/db'
+import { useDiscs, useWengine } from '@genshin-optimizer/zzz/db-ui'
+import type { TagMapNodeEntries } from '@genshin-optimizer/zzz/formula'
+import {
+  charTagMapNodeEntries,
+  conditionalEntries,
+  discTagMapNodeEntries,
+  enemy,
+  enemyDebuff,
+  ownBuff,
+  teamData,
+  wengineTagMapNodeEntries,
+  withMember,
+  withPreset,
+  zzzCalculatorWithEntries,
+} from '@genshin-optimizer/zzz/formula'
+import type { ReactNode } from 'react'
+import { useMemo } from 'react'
+
+export function CharCalcProvider({
+  character,
+  charOpt,
+  children,
+}: {
+  character: ICachedCharacter
+  charOpt: CharOpt
+  children: ReactNode
+}) {
+  const member0 = useCharacterAndEquipment(character)
+
+  const calc = useMemo(
+    () =>
+      zzzCalculatorWithEntries([
+        // Specify members present in the team
+        ...teamData([character.key]),
+        // Add actual member data
+        ...member0,
+        // TODO: Get enemy values from db
+        enemyDebuff.common.lvl.add(80),
+        enemy.common.def.add(953),
+        enemy.common.res_.electric.add(0.1),
```

Comment:
maybe remove the electric res and resRed and dmgInc and dmgRed?

Reply:
yeah i wasn't sure what was needed for calc to work